File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
def is_armstrong_number(number) -> bool:
    """Determine if a number is an Armstrong number

    An Armstrong number is a number that is the sum of its own digits, each raised
    to the number of digits of the number. This implies that all Armstrong
    numbers must be positive.
        ex: 9 is an Armstrong number: 9 -> 9^1 = 9
        ex: 10 is not an Armstrong number: 10 -> 1^2 + 0^2 != 10

    :param number:
    :type number:

    :return: True if number is an Armstrong number, False otherwise.
    """

    digits = [int(d) for d in str(number)]  # get each digit of the number
    num_digits = len(digits)
    value = sum(x ** num_digits for x in digits)  # perform Armstrong algorithm

    return number == value

File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrongnumbers

import (
	"math"
	"strconv"
	"unicode"
)

// IsNumber returns true if the number is an Armstrong number, false otherwise.
// An Armstrong Number is a number that is the sum of its own digits each to the
// power of the total number of digits
func IsNumber(n int) bool {
	asString := strconv.Itoa(n)
	numDigits := float64(len(asString))
	sum := 0

	for _, r := range asString {
		if unicode.IsDigit(r) {
			num := int(r) - int('0')
			sum += int(math.Pow(float64(num), numDigits))
		} else {
			return false
		}
	}

	return sum == n

}
